### PR TITLE
Add model comparison

### PR DESCRIPTION
**Pull request by matanper**
1. Add ModelComparisonBaseCheck
2. Separated Suite to BaseSuite and Suite
3. Add ModelComparisonSuite
4. Changed the conditions display method
5. Added show method to CheckResult and SuiteResult
6. Added auto-execute of conditions when calling `run`

### CODE REVIEW

File: deepchecks/base/display_suite.py [hunk 1]
```diff
@@ -55,13 +54,13 @@ def get_display_exists_icon(exists: bool):
...
 def display_suite_result(suite_name: str, results: List[Union[CheckResult, CheckFailure]]):
     """Display results of suite in IPython."""
-    conditions_table = []
+    checks_with_conditions = []
     display_table = []
     others_table = []
     for result in results:
         if isinstance(result, CheckResult):
             if result.have_conditions():
-                ConditionResult.append_to_conditions_table(result, conditions_table)
```
Comment (JKL98ISR) [inline deepchecks/base/display_suite.py:64]: If you sort at the end we can't keep the conditions ordered by the checks we run

**Review by JKL98ISR — CHANGES_REQUESTED** [review]

File: deepchecks/base/display_pandas.py [hunk 2]
```diff
@@ -53,14 +53,30 @@ def dataframe_to_html(df: Union[pd.DataFrame, Styler]):
...
+            sort_value = cond_result.get_sort_value()
+            icon = cond_result.get_icon()
+            header = check_result.get_header()
+            table.append([icon, header, cond_result.name, cond_result.details, sort_value])
+
+    conditions_table = pd.DataFrame(data=table,
+                                    columns=['Status', 'Check', 'Condition', 'More Info', 'sort'])
     conditions_table.sort_values(by=['sort'], inplace=True)
     conditions_table.drop('sort', axis=1, inplace=True)
+    if show_check is False:
```
Comment (JKL98ISR) [inline deepchecks/base/display_pandas.py:80]: what does it do?

File: deepchecks/base/display_suite.py (hunk 1 above)
Comment (matanper) [reply deepchecks/base/display_suite.py:64]: I don't understand the problem can you elaborate?

File: deepchecks/base/display_pandas.py (hunk 2 above)
Comment (matanper) [reply deepchecks/base/display_pandas.py:80]: what do you mean? the specific line checks if the parameter is False

File: deepchecks/base/suite.py [hunk 3]
```diff
@@ -129,45 +184,48 @@ def run(
...
-                return self
-            for c in check.checks.values():
-                self.add(c)
-        else:
-            self.checks[self._check_index] = check
-            self._check_index += 1
-        return self
+          train_datasets: A scikit-learn-compatible fitted estimator instance
+          test_datasets: Dataset object, representing data an estimator was fitted on
+          models: Dataset object, representing data an estimator predicts on
```
Comment (ItayGabbay) [inline deepchecks/base/suite.py:206]: I think there is a mismatch in the docstring here

File: deepchecks/base/display_pandas.py (hunk 2 above)
Comment (JKL98ISR) [reply deepchecks/base/display_pandas.py:80]: I got it now, so you won't show the name again

File: deepchecks/base/display_suite.py (hunk 1 above)
Comment (JKL98ISR) [reply deepchecks/base/display_suite.py:64]: oh I mistook it with another function

**Review by ItayGabbay — CHANGES_REQUESTED** [review]
A few minor comments

File: deepchecks/base/check.py [hunk 4]
```diff
@@ -232,14 +215,21 @@ def get_conditions_sort_value(self):
         """Get largest sort value of the conditions results."""
         return max([r.get_sort_value() for r in self.conditions_results])
 
+    def show(self, show_conditions=True):
+        """Call the IPython display function."""
+        self._ipython_display_(show_conditions)
```
[the remark this answers is not in the document]
Comment (matanper) [reply deepchecks/base/check.py]: outdated code

File: deepchecks/base/check.py [hunk 5]
```diff
@@ -378,6 +353,41 @@ def run(self, model) -> CheckResult:
...
+
+    def run(self,
+            train_datasets: Union[Dataset, List[Dataset]],
+            test_datasets: Union[Dataset, List[Dataset]],
+            models: List[Any]
+            ) -> CheckResult:
+        """Preprocess the parameters and pass them to `_run`."""
+        # Validations
+        if isinstance(train_datasets, Dataset) and isinstance(test_datasets, List):
+            raise DeepchecksNotSupportedError('Single train dataset with multiple test datasets is not supported.')
```
[the remark this answers is not in the document]
Comment (matanper) [reply deepchecks/base/check.py:371]: "Yet" in a class name, really?

File: deepchecks/base/check.py [hunk 6]
```diff
@@ -378,6 +353,41 @@ def run(self, model) -> CheckResult:
...
 
+class ModelComparisonBaseCheck(BaseCheck):
+    """Parent class for check that compares between two or more models."""
+
+    def run(self,
+            train_datasets: Union[Dataset, List[Dataset]],
+            test_datasets: Union[Dataset, List[Dataset]],
+            models: List[Any]
+            ) -> CheckResult:
+        """Preprocess the parameters and pass them to `_run`."""
```
[the remark this answers is not in the document]
Comment (matanper) [reply deepchecks/base/check.py:368]: feels redundant, also the rest of the base checks doesn't have

File: deepchecks/base/check.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ItayGabbay) [reply deepchecks/base/check.py:371]: OK, you win

**Review by ItayGabbay — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**